### PR TITLE
Adds minimal example; minor refactor

### PR DESCRIPTION
**Pull request by vwxyzjn**
Some minor refactor

### CODE REVIEW

File: open_instruct/finetune.py [hunk 1]
```diff
@@ -451,28 +451,27 @@ def main(args: FlatArguments):
...
+
+    # debugging tool for fewer samples
+    if args.max_train_samples is not None:
+        max_train_samples = min(len(train_dataset), args.max_train_samples)
+        logger.info(f"Limiting training samples to {max_train_samples} from {len(train_dataset)}.")
+        train_dataset = train_dataset.select(range(max_train_samples))
+
     with accelerator.main_process_first():
-        lm_datasets = raw_datasets.map(
+        train_dataset = train_dataset.map(
```
Comment (vwxyzjn) [inline open_instruct/finetune.py:463]: In `finetune.py` the other splits are never really used, so I just replacing with train_dataset

File: open_instruct/finetune.py [hunk 2]
```diff
@@ -451,28 +451,27 @@ def main(args: FlatArguments):
     else:
         raise ValueError("You need to have either 'prompt'&'completion' or 'messages' in your column names.")
 
+    train_dataset = raw_datasets["train"]
+
+    # debugging tool for fewer samples
+    if args.max_train_samples is not None:
+        max_train_samples = min(len(train_dataset), args.max_train_samples)
+        logger.info(f"Limiting training samples to {max_train_samples} from {len(train_dataset)}.")
+        train_dataset = train_dataset.select(range(max_train_samples))
```
Comment (vwxyzjn) [inline open_instruct/finetune.py:460]: Saves some time by moving up `train_dataset.select(range(max_train_samples)` before any tokenization (which is slowwwww)

File: scripts/README.md [hunk 3]
```diff
@@ -16,17 +16,23 @@ For Ai2 users, these scripts all work best in interactive sessions (not in batch
 2. `finetune_qlora_with_acceralate.sh`: Script for running `open_instruct/finetune.py` with QLoRA.
 3. `finetune_with_acceralate_config.sh`: Script for running `open_instruct/finetune.py` with configs found in `configs/train_configs/sft/`. Good for reproducing results. Example usages:
 
-```
+```bash
+sh scripts/finetune_with_accelerate_config.sh 1 configs/train_configs/sft/mini.yaml
```
Comment (vwxyzjn) [inline scripts/README.md:20]: Added minimal example that can be done in like 3 mins.

**Review by hamishivi — APPROVED** [review]
Looks good to me, thanks for cleaning up my changes 😅

**Pull request merged**